kernel/poll.c

```diff
@@ -46,14 +46,24 @@ dword_t sys_select(fd_t nfds, addr_t readfds_addr, addr_t writefds_addr, addr_t
         return _EFAULT;
 
     struct timespec timeout_ts = {};
-    if (timeout_addr != 0) {
+    if (timeout_addr != 0 && timeout_ts_addr != 0) {
+        assert(!"select_internal BUG");
```

Comment:
We can prevent this error at compile time instead of runtime. Let's move the code to read and convert timeval to the caller, removing the need for two mutually exclusive arguments. There are also [utility functions](https://github.com/ish-app/ish/blob/33eaf3bd2c0402309310b6b518edda2300c9df61/kernel/fs.c#L735C1-L747C2) already written for similar purposes, we should move them somewhere like util/timer.h and use them.

Reply:
Done.